pydantic_ai_slim/pydantic_ai/_cli.py

```diff
@@ -0,0 +1,170 @@
+        except KeyboardInterrupt:
+            pass
+        return 0
+
+    history = Path().home() / '.prompt-history.txt'
+    session = PromptSession(history=FileHistory(str(history)))  # type: ignore
+    multiline = False
+    messages: list[ModelMessage] = []
+
+    while True:
+        try:
+            text = cast(str, session.prompt('pydai ➤ ', auto_suggest=AutoSuggestFromHistory(), multiline=multiline))
+        except (KeyboardInterrupt, EOFError):
+            return 0
+
+        if not text.strip():
+            continue
+
+        ident_prompt = text.lower().strip(' ').replace(' ', '-')
+        if ident_prompt == 'multiline':
+            multiline = not multiline
+            if multiline:
+                console.print(
+                    'Enabling multiline mode. '
+                    '[dim]Press [Meta+Enter] or [Esc] followed by [Enter] to accept input.[/dim]'
+                )
+            else:
+                console.print('Disabling multiline mode.')
+            continue
+
+        try:
+            messages = asyncio.run(ask_agent(agent, text, stream, console, messages))
+        except KeyboardInterrupt:
+            return 0
+
+
+async def ask_agent(
+    agent: Agent,
+    prompt: str,
+    stream: bool,
+    console: Console,
+    messages: list[ModelMessage] | None = None,
+) -> list[ModelMessage]:
+    async with agent.iter(prompt, message_history=messages) as agent_run:
+        console.print('\nResponse:', style='green')
+
+        content: str = ''
+        interrupted = False
+        with Live('', refresh_per_second=15, console=console) as live:
+            try:
+                async for node in agent_run:
+                    if Agent.is_model_request_node(node):
+                        async with node.stream(agent_run.ctx) as handle_stream:
+                            async for event in handle_stream:
+                                if isinstance(event, PartDeltaEvent) and isinstance(event.delta, TextPartDelta):
+                                    if stream:
+                                        content += event.delta.content_delta
+                                        live.update(Markdown(content))
+            except KeyboardInterrupt:
+                interrupted = True
+
+        if interrupted:
+            console.print('[dim]Interrupted[/dim]')
+
+        assert agent_run.result
+        if not stream:
+            content = agent_run.result.data
+            console.print(Markdown(content))
+        return agent_run.result.all_messages()
+
+
+class Provider(TypedDict):
+    name: Literal['openai']
+    api_key: str
+
+
+def infer_provider() -> Provider:
+    """Infer the provider from the environment variables."""
+    try:
+        openai_api_key = os.environ['OPENAI_API_KEY']
+    except KeyError:
+        raise ValueError('Only OpenAI is supported at the moment, please set the OPENAI_API_KEY environment variable.')
+    return {'name': 'openai', 'api_key': openai_api_key}
```

Comment:
Is this now unused? I feel like it shouldn't be necessary since you can specify the provider via command line argument.

I'll note that I still think we should support specifying the model via an environment variable. E.g. `PYDAI_MODEL` or similar